packages/get-platform/src/getPlatform.ts

```diff
@@ -206,5 +239,36 @@ export async function getPlatform(): Promise<Platform> {
   }
 
   // use the debian build with OpenSSL 1.1 as a last resort
+  // TODO: perhaps we should default to 'debian-openssl-3.0.x'
   return 'debian-openssl-1.1.x'
 }
+
+/**
+ * Given a promise generator, returns the promise's result.
+ * If the promise throws, returns undefined.
+ */
+async function discardError<T>(runPromise: () => Promise<T>): Promise<T | undefined> {
+  try {
+    return await runPromise()
+  } catch (e) {
+    return undefined
+  }
+}
+
+/**
+ * Given a list of system commands, returns the first successful command's stdout.
```

Comment:
I'd clarify that "first" means first in the order in the input array, not the one that completed first.